be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/ElectionHandler.scala

```diff
@@ -108,7 +108,7 @@ class ElectionHandler(dbRef: => AskableActorRef) extends MessageHandler {
     }
     val electionChannel: Channel = rpcMessage.getParamsChannel
     val combined = for {
-      electionQuestionResults <- createElectionQuestionResults(electionChannel, rpcMessage.extractLaoId)
+      electionQuestionResults <- createElectionQuestionResults(electionChannel, electionChannel.decodeChannelLaoId.get)
```

Comment:
Be careful here. In case you have a empty LAO id, it will throw an exception. 
How can you deal with that ?

Reply:
Also, maybe it would be more clear if you define a value for the laoId

Reply:
Okeey I realize now that we forgot to speak about this during the meeting.. 

So, your implementation is actually good ! But you can do something more concise, and actually you have a already-implemented function that you can use.. 
Remember the function `extractParameters` in the `MessageHandler.scala` ? There is a similar function there that can really help you for this. :)